sw/host/opentitanlib/src/tpm/driver.rs

```diff
@@ -254,14 +293,33 @@ impl I2cDriver {
 
 impl Driver for I2cDriver {
     fn read_register(&self, register: Register, data: &mut [u8]) -> Result<()> {
-        self.i2c.run_transaction(
-            self.addr,
-            &mut [
-                i2c::Transfer::Write(&[Self::addr(register).unwrap()]),
-                i2c::Transfer::Read(data),
-            ],
-        )?;
-        Ok(())
+        // Retry in case the I2C bus wasn't ready.
+        for i in 0..10 {
+            let res = self.i2c.run_transaction(
+                self.addr,
+                &mut [
+                    i2c::Transfer::Write(&[Self::addr(register).unwrap()]),
+                    i2c::Transfer::Read(data),
+                ],
+            );
+            match res {
+                Err(e) => {
+                    log::error!(
+                        "Register 0x{:X} access error: {}",
+                        Self::addr(register).unwrap(),
+                        e
+                    );
+                }
+                Ok(()) => {
+                    if i > 0 {
+                        log::error!("Success after {} tries.", i);
+                    }
+                    return Ok(());
+                }
+            }
+            thread::sleep(Duration::from_millis(100));
+        }
+        bail!("Failed to read TPM register.");
```

Comment:
Rather than bailing or returning a formal error here, it might be better to return from the `Err` branch of the match on the last iteration of the loop, thus returning the actual transaction error.